cmd/root.go

```diff
@@ -115,63 +121,76 @@ func init() {
 	rootCmd.PersistentFlags().MarkHidden("send-random-data-ext")
 }
 
-func detectNewVersion() {
-	const versionMetadataUrl = "https://aka.ms/azcopyv10-version-metadata"
+// always spins up a new goroutine, because sometimes the aka.ms URL can't be reached (e.g. a constrained environment where
+// aka.ms is not resolvable to a reachable IP address). In such cases, this routine will run for ever, and the caller should
```

Comment:
If aka.ms is not resolvable, shouldn't there be a network error and we'd retry it? Why is it stuck forever?

Reply:
Good point. Maybe I just gave up before the finite timeout. But it was long enough to fool both me and the user who logged it into thinking it was "for ever"